### PR TITLE
feat(dbt): allow runtime compilation of manifest in Dagster scaffold

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary & Motivation
We essentially provide a workflow that emulates the behavior of `load_assets_from_dbt_project`. In this workflow, the dbt manifest is created at runtime, and is piped into `@dbt_assets`.

Users can invoke this flow when developing by doing:

```bash
DAGSTER_DBT_PARSE_PROJECT_ON_LOAD=1 dagster dev
```

## How I Tested These Changes
pytest, local

### CODE REVIEW

**Comment by rexledesma** [comment]
Current dependencies on/for this PR:
* master
  * **PR #15318** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15318" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/15318?utm_source=stack-comment).

File: python_modules/libraries/dagster-dbt/dagster_dbt/include/scaffold/definitions.py.jinja [hunk 1]
```diff
@@ -7,6 +7,14 @@ from dagster_dbt import DbtCli, dbt_assets
 dbt_project_dir = Path(__file__).parent.joinpath({{ dbt_project_dir_relative_path_parts | join(', ')}})
 dbt_manifest_path = dbt_project_dir.joinpath("target", "manifest.json")
 
+dbt = DbtCli(project_dir=os.fspath(dbt_project_dir))
+
+if os.getenv("DAGSTER_DBT_BUILD_PROJECT"):
+    dbt_parse_invocation = dbt.cli(["parse"], manifest={})
+    dbt_parse_invocation.wait()
+    dbt_parse_invocation.target_path.joinpath("manifest.json").replace(
```
Comment (sryza) [inline python_modules/libraries/dagster-dbt/dagster_dbt/include/scaffold/definitions.py.jinja:15]: What exactly is this doing / for?

File: python_modules/libraries/dagster-dbt/dagster_dbt/include/scaffold/definitions.py.jinja [hunk 2]
```diff
@@ -7,6 +7,14 @@ from dagster_dbt import DbtCli, dbt_assets
 dbt_project_dir = Path(__file__).parent.joinpath({{ dbt_project_dir_relative_path_parts | join(', ')}})
 dbt_manifest_path = dbt_project_dir.joinpath("target", "manifest.json")
 
+dbt = DbtCli(project_dir=os.fspath(dbt_project_dir))
+
+if os.getenv("DAGSTER_DBT_BUILD_PROJECT"):
+    dbt_parse_invocation = dbt.cli(["parse"], manifest={})
```
Comment (sryza) [inline python_modules/libraries/dagster-dbt/dagster_dbt/include/scaffold/definitions.py.jinja:13]: We theoretically don't need the manifest argument here, right? I filed https://github.com/dagster-io/dagster/issues/15233 to track.

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/cli/test_app.py [hunk 3]
```diff
@@ -77,6 +77,54 @@ def test_project_scaffold_command(
     assert materialize_dbt_models_schedule.cron_schedule == "0 0 * * *"
 
 
+def test_project_scaffold_command_with_runtime_manifest(
```
Comment (sryza) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/cli/test_app.py:80]: Nice test

File: python_modules/libraries/dagster-dbt/dagster_dbt/include/scaffold/definitions.py.jinja (hunk 1 above)
Comment (rexledesma) [reply python_modules/libraries/dagster-dbt/dagster_dbt/include/scaffold/definitions.py.jinja:15]: We are moving the manifest that's generated in the dynamic target path of a `DbtCli.cli` invocation (e.g. `target-123213`) to the static target path of the dbt project `target/`.

Comment (sryza) [reply python_modules/libraries/dagster-dbt/dagster_dbt/include/scaffold/definitions.py.jinja:15]: Gotcha. That sounds correct, but the dataflow is a little hard to follow, and it overwrites the manifest at the static target path which may or may not be desirable. What if we made something like this possible?

```python
if os.getenv("DAGSTER_DBT_BUILD_PROJECT"):
    dbt_manifest_path = dbt.cli(["parse"]).wait().get_output_manifest_path()
else:
    dbt_manifest_path = dbt_project_dir.joinpath("target", "manifest.json")
```
